src/Viz.js

```diff
@@ -337,11 +342,9 @@ export default class Viz extends BaseClass {
     if (this.legendPosition() === "top" || this.legendPosition() === "bottom") drawLegend.bind(this)(this._filteredData);
     if (this.colorScalePosition() === "top" || this.colorScalePosition() === "bottom") drawColorScale.bind(this)(this._filteredData);
 
-    this._shapes = [];
```

Comment:
sorry I missed this on the last review: we can't delete this line, as the `this._shapes` variable needs to be reset to empty on each redraw. 

additionally, please see my previous comment about pulling from master to fix the merge conflict.

Reply:
@rbaheti I still need you to fix this issue.

Reply:
Oh no! I thought I fixed it. :(